### PR TITLE
Move SparseArrays to an extension

### PR DESCRIPTION
**Pull request by jishnub**

### CODE REVIEW

**Comment by dlfivefifty** [comment]
Could you add tests for those two lines? (I realise they weren't covered before but just to make sure it isn't breaking something that was working)

**Comment by jishnub** [comment]
I think it's almost a weak dep, but I found that `FillArrays.axes_print_matrix_row` is used in `Base.print_matrix_row` without any reference to `AbstractFill` types

**Comment by dkarrasch** [comment]
How many `methods` exist for `axes_print_matrix_row` in `FillArrays.jl`? Do we need to dispatch, or could we simply have a copy of that function here? If it doesn't dispatch on `FillArrays.jl` types, it should work for `ArrayLayouts.jl`, using Base or ArrayLayouts.jl-owned methods and types, shouldn't it?

**Comment by dkarrasch** [comment]
Actually, this seems to be an easy case. `FillArrays.jl` defines

```julia
if VERSION < v"1.8-"
    axes_print_matrix_row(lay, io, X, A, i, cols, sep) =
        Base.invoke(Base.print_matrix_row, Tuple{IO,AbstractVecOrMat,Vector,Integer,AbstractVector,AbstractString},
                    io, X, A, i, cols, sep)
else
    axes_print_matrix_row(lay, io, X, A, i, cols, sep, idxlast::Integer=last(axes(X, 2))) =
        Base.invoke(Base.print_matrix_row, Tuple{IO,AbstractVecOrMat,Vector,Integer,AbstractVector,AbstractString,Integer},
                    io, X, A, i, cols, sep, idxlast)
end
```

so it is just an indirection, without reference to `FillArrays`'s methods or types, which, on top, can be removed altogether once the lower bound is lifted to some version higher than v1.8. So `axes_print_matrix_row` could be copied here, and the `import` removed. If that solves the dependency issue already, that would be awesome!

File: ext/ArrayLayoutsSparseArraysExt.jl [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+using ArrayLayouts
+using ArrayLayouts: _copyto!
+using SparseArrays
+import LinearAlgebra
+
+import Base: copyto!
+
+# ambiguity from sparsematrix.jl
+copyto!(dest::LayoutMatrix, src::SparseArrays.AbstractSparseMatrixCSC) =
```
Comment (dlfivefifty) [inline ext/ArrayLayoutsSparseArraysExt.jl:11]: Can you add a test for this line?

**Review by dlfivefifty — CHANGES_REQUESTED** [review]

File: ext/ArrayLayoutsSparseArraysExt.jl (hunk 1 above)
Comment (jishnub) [reply ext/ArrayLayoutsSparseArraysExt.jl:11]: I've added a test downstream in `BandedMatrices`, so this line is now covered. However, for some reason, the test fails on the CI, although it passes locally for me. Haven't had time to look into it yet

Comment (jishnub) [reply ext/ArrayLayoutsSparseArraysExt.jl:11]: Oh, I was looking at the tests for the other PR. The tests do pass for this one, but the coverage isn't being updated for some reason

**Comment by jishnub** [comment]
The patch coverage is fixed now

**Comment by dkarrasch** [comment]
So, shall we also make FillArrays.jl a weak dependency here? Can I help with anything?

**Comment by jishnub** [comment]
`FillArrays` also uses `axes_print_matrix_row` in `print_matrix_row`. Would it be possible to resolve that one as well?
https://github.com/JuliaArrays/FillArrays.jl/blob/a718e2f66b9a4ddb29a5bd8539cdf127b2280c22/src/FillArrays.jl#L756-L764

**Comment by dkarrasch** [comment]
Sure, you just keep things internal, don't import them from each other, and have copies in each of the packages. And, TBH, that's what they are: internal functions. One could in fact get rid of them completely by having version-dependent versions of the print functions, which call the Base functions directly.

**Comment by jishnub** [comment]
You are welcome to submit a PR with the changes

**Comment by jishnub** [comment]
@dlfivefifty should be good to merge? Coverage is fixed now

**Pull request merged**